content/languages/ruby/authoring.md

```diff
@@ -0,0 +1,156 @@
+---
+kind: tutorial
+languages: [ruby]
+sidebar: "language:ruby"
+prev: /languages/ruby/codewars-test/
+---
+
+# Ruby: creating and translating a kata
+
+This article is intended for kata authors and translators who would like to create new content in Ruby. It attempts to explain how to create and organize things in a way conforming to [authoring guidelines](/authoring/guidelines/), shows the most common pitfalls, and how to avoid them.
+
+This article is not a standalone tutorial on creating kata or translations. It's meant to be a complementary, Ruby-specific part of a more general set of HOWTOs and guidelines related to [content authoring](/authoring/). If you are going to create a Ruby translation or a new Ruby kata from scratch, please make yourself familiar with the previously mentioned documents related to authoring in general first. 
+
+## General info
+
+Any technical information related to the Ruby setup on Codewars can be found on the [Ruby reference](/languages/ruby/) page (language versions, available gems, and setup of the code runner).
+
+## Description
+
+Ruby code blocks can be inserted with Ruby-specific part in [sequential code blocks](/references/markdown/extensions/#sequential-code-blocks):
+
+~~~
+```ruby
+
+...your code here...
+
+```
+~~~
+
+Ruby-specific paragraphs can be inserted with [language conditional rendering](/references/markdown/extensions/#conditional-rendering):
+
+```
+~~~if:ruby
+
+...text visible only for Ruby description...
+
+~~~
+
+~~~if-not:ruby
+
+...text not visible in Ruby description...
+
+~~~
+```
+
+## Tasks and Requirements
+
+Some concepts don't always translate well to or from Ruby. Because of this, some constructs should be avoided and some translations just shouldn't be done.
+- Avoid returning different data types depending on the situation (_"Return the result, or the string 'Error' if no result can be found..."_). Ruby is dynamically typed, which is not the case for some other languages. Returning `nil` might be appropriate in some situations, but throwing an error might be better in others.
+
+Some kata just should not be translated into Ruby because it can be difficult to keep their initial idea:
+- Some kata might be meant for another language specifically. For example, a kata about `Python: Learning Classes` should probably not be translated to Ruby, even though Ruby has `class` syntax.
```

Comment:
I don't feel like it's evident. I practice both (well, not much ruby, yeah, but still) and I honestly don't see why thsi should be avoided in ruby => ?

Reply:
You probably could, but you'd want to change the kata title at least, right?

Reply:
oh... But it's only about the title, I feel the reason is out of place, then (but maybe it's just me being oriented by what I knew will follow...?)

Reply:
I think I just need a better example of something that shouldn't... any ideas?

Reply:
maybe something what does not translate well into hashes, or duck typing?

Reply:
You can technically do duck typing in Ruby...
Maybe like something about JS coercion? Ruby doesn't have coercion (yay).

Reply:
Damn I am so noob in Ruby I have no idea what could apply here.

Maybe dudner methods? Coercion sounds good too.


 Maybe just leave this out?
 
 
I KNOW! Performance kata do not translate well into Ruby! :D

Reply:
Dunder methods sounds right.

Reply:
don't ruby classes have `def coerce(self,other)` mehtods? x) 

even dunder methods are a bit edgy because some exist in ruby `=, +, ...`

maybe step out of python, for this? like "learn the difference int/long" in java/C/... doesn't translate to ruby.

?